File: src/PRMS.Domain/Entities/Product.cs
using PRMS.Domain.Enums;

namespace PRMS.Domain.Entities;

public class Product : BaseEntity
{
    public string ProductCode { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    
    public ProductCategory Category { get; set; }
    public string? SubCategory { get; set; }
    
    public string? PartNumber { get; set; }
    public string? Manufacturer { get; set; }
    public string? Brand { get; set; }
    
    public string Unit { get; set; } = "EA";
    public decimal? StandardPrice { get; set; }
    
    public int? MinOrderQuantity { get; set; }
    public int? MaxOrderQuantity { get; set; }
    public int? LeadTimeDays { get; set; }
    
    public bool IsActive { get; set; }
    public bool RequiresApproval { get; set; }
    
    public string? ImageUrl { get; set; }
    public string? Specification { get; set; }
    
    public virtual ICollection<PriceHistory> PriceHistory { get; set; } = new List<PriceHistory>();
    public virtual ICollection<VendorProduct> VendorProducts { get; set; } = new List<VendorProduct>();
}

File: frontend/src/shared/types/product.ts
import type { BaseEntity } from './common';

export type ProductCategory = 
  | 'Electronics'
  | 'Furniture'
  | 'Supplies'
  | 'Equipment'
  | 'Software'
  | 'Services'
  | 'Other';

export interface ProductResponse extends BaseEntity {
  productCode: string;
  name: string;
  description?: string;
  category: ProductCategory;
  subCategory?: string;
  partNumber?: string;
  manufacturer?: string;
  brand?: string;
  unit: string;
  standardPrice: number;
  minOrderQuantity: number;
  maxOrderQuantity?: number;
  leadTimeDays: number;
  isActive: boolean;
  requiresApproval: boolean;
  imageUrl?: string;
  specification?: string;
}

export interface CreateProductDto {
  productCode: string;
  name: string;
  description?: string;
  category: string;
  subCategory?: string;
  partNumber?: string;
  manufacturer?: string;
  brand?: string;
  unit: string;
  standardPrice: number;
  minOrderQuantity: number;
  maxOrderQuantity?: number;
  leadTimeDays: number;
  isActive: boolean;
  requiresApproval: boolean;
  imageUrl?: string;
  specification?: string;
}

export interface UpdateProductDto extends CreateProductDto {
  id: string;
}

export interface ProductFilters {
  category?: ProductCategory | string;
  search?: string;
  isActive?: boolean;
}
